libs/core/src/lib/resizable-card-layout/resizable-card-item/resizable-card-item.component.ts

```diff
@@ -0,0 +1,588 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    EventEmitter,
+    HostBinding,
+    HostListener,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output,
+    ViewEncapsulation
+} from '@angular/core';
+import { FocusableOption } from '@angular/cdk/a11y';
+import { Subject } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+
+export type ResizeDirection = 'vertical' | 'horizontal' | 'both';
+
+// Card resizes in step of fixed values. values are in pixel
+export const HorizontalResizeStep = 320;
+export const verticalResizeStep = 16;
+export const gap = 16;
+export const horizontalResizeOffset = 20;
+export const verticalResizeOffset = 10;
+
+let cardRank = 1;
+let cardUniqueId = 0;
+
+@Component({
+    selector: 'fd-resizable-card-item',
+    templateUrl: 'resizable-card-item.component.html',
+    styleUrls: ['./resizable-card-item.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class ResizableCardItemComponent implements OnInit, OnDestroy, FocusableOption {
+    /** Set card properties from the config received */
+    @Input()
+    get config(): ResizableCardItemConfig {
+        return this._config;
+    }
+
+    set config(config: ResizableCardItemConfig) {
+        this._config = config;
+        this._initialSetup();
+        this._cd.detectChanges();
+    }
+
+    /** Item id attribute value */
+    @Input()
+    itemId = `fd-card-item-${cardUniqueId++}`;
+
+    /** card title */
+    @Input()
+    title: string;
+
+    /** Position rank  of card */
+    @Input()
+    rank: number = cardRank++;
```

Comment:
If user sets the `rank` on another card, `cardRank` may not always be the maximum value. You might want to use setter-getter here and update `cardRank` to be the maximum value whenever  `rank` is set.

Reply:
fixed now

Reply:
fixed now